python/cudf/cudf/pandas/fast_slow_proxy.py

```diff
@@ -874,7 +889,23 @@ def __name__(self, value):
         setattr(self._fsproxy_slow, "__name__", value)
 
 
-def _fast_slow_function_call(func: Callable, /, *args, **kwargs) -> Any:
+def _assert_fast_slow_eq(left, right, **kwargs):
+    assert_func = (
+        assert_eq
+        if not kwargs.get("assert_func")
+        else kwargs.get("assert_func")
+    )
+    if type(left).__name__ in _TYPES:
+        assert_func(left, right)
+
+
+def _fast_slow_function_call(
+    func: Callable,
+    debug_mode: str | None = None,
+    assert_func: Callable | None = None,
```

Comment:
Right now the assert function and the debug mode are module-scope variables that are passed through everywhere. We don't need both. If you think one or both makes sense to be overridable as a parameter, then I would just defer determination of whether to use the default or not to the call site of `_assert_fast_slow_eq`. If you always intend to end up with the module-scope versions, then remove the parameters and just use it directly. The current approach adds complexity without any real benefits I can see.